src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -312,27 +426,25 @@ def find_fixed_range(testcase_id, job_type):
     unsymbolized_crash_stacktrace = result.get_stacktrace(symbolized=False)
     stacktrace = utils.get_crash_stacktrace_output(
         command, symbolized_crash_stacktrace, unsymbolized_crash_stacktrace)
-    testcase = data_handler.get_testcase_by_id(testcase_id)
-    testcase.last_tested_crash_stacktrace = data_handler.filter_stacktrace(
-        stacktrace)
-    data_handler.update_progression_completion_metadata(
-        testcase,
-        max_revision,
-        is_crash=True,
-        message='still crashes on latest revision r%s' % max_revision)
 
-    # Since we've verified that the test case is still crashing, clear out any
-    # metadata indicating potential flake from previous runs.
-    task_creation.mark_unreproducible_if_flaky(testcase, False)
-
-    # For chromium project, save latest crash information for later upload
-    # to chromecrash/.
     state = result.get_symbolized_data()
-    crash_uploader.save_crash_info_if_needed(testcase_id, max_revision,
-                                             job_type, state.crash_type,
-                                             state.crash_address, state.frames)
-    return
+    testcase.last_tested_crash_stacktrace = data_handler.filter_stacktrace(
+        stacktrace)
 
+    # Pass these back to the trusted worker through the testcase.    
+    testcase.crash_type = state.crash_type
+    testcase.crash_address = state.crash_address
+    testcase.crash_state = state.frames
+    testcase.crash_revision = max_revision
+    error_message = 'still crashes on latest revision r%s' % max_revision
+    return uworker_io.UworkerOutput(
+          testcase=testcase,
+          uworker_input=uworker_input,
+          error_message=error_message,
+          error=uworker_msg_pb2.ErrorType.PROGRESSION_CRASH_ON_LATEST)  
+
+  # TODO: these don't seem to be used anywhere.
```

Comment:
@oliverchang I think crashes_on_unexpected_state is not used anywhere. Can we get rid of it?
Ali I guess if we get rid of it, we shoudl get rid of result.unexpected_crash as it isn't used for any other purpose.